### ISSUE TITLE
cp --from-fs --to-bucket

### ISSUE THREAD

**Issue by asilvas** [generalui/s3p#31]
Would be super handy for transferring data into S3.

**Comment by shanebdavis** [comment]
Yes! This is definitely on my list. You'd think it would be easy... but the way S3P does it's massively parallel bucket-scanning doesn't translate very well to the local file system. I was able to make a workable solution from-bucket-to-fs, but it's harder the other way around.

Still, it can and should be done!

**Comment by mbolek** [comment]
> I was able to make a workable solution from-bucket-to-fs, but it's harder the other way around.

Could it be made available? :)

**Comment by shanebdavis** [comment]
There is a 'quick and dirty' way to make this happen, but it has scaling limitations. Basically, I could load the list of all the source files to copy in memory and the simulate the S3 listBucket API against that list. 

Would it be helpful if this was available, but it might blow up if you have too many local files? I'm guessing under 100,000 would be no problem. Possibly even under a million files would work. More than a million files might start to get out of control.

@moblek @asilvas What do you think?
